saw-remote-api/python/saw_client/crucible.py

```diff
@@ -604,18 +622,21 @@ def to_json(self) -> Any:
 
 # It's tempting to name this `global` to mirror SAWScript's `llvm_global`,
 # but that would clash with the Python keyword `global`.
-def global_var(name: str) -> SetupVal:
+def global_var(name: str) -> GlobalVarVal:
     """Returns a pointer to the named global ``name`` (i.e., a ``GlobalVarVal``)."""
     return GlobalVarVal(name)
 
-# FIXME Is `Any` too permissive here -- can we be a little more precise?
-def cryptol(data : Any) -> 'CryptolTerm':
-    """Constructs a Cryptol value from ``data`` (i.e., a ``CryptolTerm``, which is also a ``SetupVal``).
+def cry(s : str) -> CryptolTerm:
+    """Like ``cry`` from the Cryptol bindings, but retuns a ``CryptolTerm``
+    (which is also a ``SetupVal``)."""
+    return CryptolTerm(s)
 
-    ``data`` should be a string literal representing Cryptol syntax or the result of a Cryptol-realted server computation."""
-    return CryptolTerm(data)
+def cry_f(s : str) -> CryptolTerm:
+    """Like ``cry_f`` from the Cryptol bindings, but retuns a ``CryptolTerm``
+    (which is also a ``SetupVal``)."""
+    return CryptolTerm(to_cryptol_str_customf(s, frames=1, filename='<cryptol_f>'))
 
-def array(*elements: SetupVal) -> SetupVal:
+def array(*elements: SetupVal) -> ArrayVal:
```

Comment:
Does knowing it's a Python `ArrayVal` buy us something here?

E.g., if there are actually Python methods that take a `ArrayVal`, then having this return that more specific type I think makes a lot of sense. If our API is actually all written in terms of `SetupVal`, I think it might be more clear to just stick with `SetupVal`. Anyway, that's just my two bits on this.

Reply:
Same for other more specific return types.